unity-client/Assets/Scripts/MainScripts/DCL/Controllers/AssetManager/Common/AssetPromiseKeeper.cs

```diff
@@ -245,6 +290,11 @@ private void LoadBlockedPromises(object loadedPromiseId)
                 promise.library = library;
                 promise.OnPreFinishEvent += CleanPromise;
```

Comment:
has everything been pushed? i think un your computer, locally, this `CleanPromise()` was being called there without subscribing it to the `promise.OnPreFinishEvent`, is this OK here?

Reply:
I found a cleaner way of implementing the fix, but in any case both approached missed a new case that I discovered today. Just added a few changes that should address all cases.